restartmanager/restartmanager.go

```diff
@@ -72,8 +75,12 @@ func (rm *restartManager) ShouldRestart(exitCode uint32, hasBeenManuallyStopped
 	}
 	if rm.timeout == 0 {
 		rm.timeout = defaultTimeout
-	} else {
+	} else if rm.timeout > maxRestartTimeout {
+		return false, nil, fmt.Errorf("reached max restart timeout")
```

Comment:
`rm.timeout = maxRestartTimeout`

We still want the container to be restarted every houru